lightning/src/ln/channelmanager.rs

```diff
@@ -4662,6 +4781,33 @@ where
 				should_persist = NotifyOption::DoPersist;
 			}
 
+			// Close remaining channels in funding batches when one channel closes.
```

Comment:
This is done to simplify locking behavior: `update_maps_on_chan_removal` marks the batch for closure while holding a lock on the closing channel. We would likely never want to hold locks across multiple channels at the same time.

Reply:
Ah, specifically, the lock of `PeerState` is a `Mutex`?

Reply:
Unless I missed anything, at any point in time, held locks were acquired in the following order:
1. `per_peer_state` locks.
2. `funding_batch_states` locks.
3. `funding_batches_to_be_closed` locks.

If a lock is being released, we might for example acquire the `funding_batches_to_be_closed` lock, release it, and then acquire the `funding_batch_states` lock, but the above should still be valid for any point in the code.

For example, below we acquire the `funding_batch_states` and `funding_batches_to_be_closed` locks in that order, release both, and then acquire the `per_peer_state` lock.

Reply:
> Ah sorry I thought you were referring to just the order of the different per_peer_state locks.

Ah you're right, sorry I lost track there. But yes, similarly, we will never hold two `peer_state` locks at the same time I believe? The main scenario I wanted to avoid with this setup is a circular wait between `peer_state` locks. For example, say you have a batch containing a channel with peer A and a channel with peer B. If both peers disconnect at the same time and `update_maps_on_chan_removal` is responsible for propagating closures while holding the `peer_state` lock, it is possible that you will have a lock on `peer_state` A and try to acquire a lock on `peer_state` B at the same time as another task is having a lock on `peer_state` B and trying to acquire `peer_state` A. This is assuming that the locks won't be released in `update_maps_on_chan_removal` but that would indeed require a larger refactor.